test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,343 @@
+import { expect } from "chai";
+import hre, { ethers, deployments } from "hardhat";
+import { getSafeWithSingleton, getSafeSingletonAt, safeMigrationContract } from "../utils/setup";
+import deploymentData from "../json/safeDeployment.json";
+import safeRuntimeBytecode from "../json/safeRuntimeBytecode.json";
+import { executeContractCallWithSigners } from "../../src/utils/execution";
+
+const SAFE_SINGLETON_ADDRESS = "0x477C3fb2D564349E2F95a2EF1091bF9657b26145";
+
+const SAFE_SINGLETON_L2_ADDRESS = "0x551A2F9a71bF88cDBef3CBe60E95722f38eE0eAA";
+
+const COMPATIBILITY_FALLBACK_HANDLER = "0x4c95c836D31d329d80d696cb679f3dEa028Ad4e5";
+
+const FALLBACK_HANDLER_STORAGE_SLOT = "0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5";
+
+const GUARD_STORAGE_SLOT = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
+
+const migrationPaths = [
```

Comment:
Nice organized way to write it like this!

Should we also have one for `1.4.1 > 1.5.0`?

Reply:
Yes

Reply:
https://github.com/safe-global/safe-smart-account/pull/793/commits/8c2d8016dbcb48d9e840de2b0956422984ea86e0